polynote-frontend/polynote/ui/component/notebook/cell.ts

```diff
@@ -609,6 +610,34 @@ abstract class Cell extends Disposable {
             .when("RunToCursor", () => {
                 this.dispatcher.runToActiveCell()
             })
+            .when("MoveAbove", () => {
+                this.notebookState.moveCell("above");
+                return ["stopPropagation", "preventDefault"]
+            })
+            .when("MoveBelow", () => {
+                this.notebookState.moveCell("below");
+                return ["stopPropagation", "preventDefault"]
+            })     
+            .when("Merge", () => {
```

Comment:
This is a tricky one. What should happen if someone tries to merge a Scala and Python cell together? Is there any way to undo this change easily? 

Thoughts @jeremyrsmith ?

Reply:
We don't have a global undo buffer, currently. We just rely on Monaco for text (and a one-off thing for undoing cell deletion).

So we'd have to add a global undo mechanism first (and I suppose the "Click to undo cell deletion" thing would go away and get rolled into that?)

Reply:
But generally I don't think it should let you merge cells of different languages. Should just do nothing if you attempt to.